cli/src/helpers/installDependencies.ts

```diff
@@ -1,27 +1,76 @@
 import chalk from "chalk";
 import { execa } from "execa";
-import ora from "ora";
-import { getUserPkgManager } from "~/utils/getUserPkgManager.js";
+import ora, { Ora } from "ora";
+import {
+  getUserPkgManager,
+  PackageManager,
+} from "~/utils/getUserPkgManager.js";
 import { logger } from "~/utils/logger.js";
 
 type Options = {
   projectDir: string;
 };
 
+/*eslint-disable @typescript-eslint/no-floating-promises*/
+const runInstallCommand = async (
+  pkgManager: PackageManager,
+  projectDir: string,
+): Promise<Ora | null> => {
+  switch (pkgManager) {
+    // When using npm, inherit the stderr stream so that the progress bar is shown
+    case "npm":
+      await execa(pkgManager, ["install"], {
+        cwd: projectDir,
+        stderr: "inherit",
+      });
+
+      return null;
+    // When using yarn or pnpm, use the stdout stream and ora spinner to show the progress
+    case "pnpm":
+      const pnpmSpinner = ora("Running pnpm install...").start();
+      const pnpmSubprocess = execa(pkgManager, ["install"], {
+        cwd: projectDir,
+        stdout: "pipe",
+      });
+
+      await new Promise<void>((res, rej) => {
+        pnpmSubprocess.stdout?.on("data", (data: Buffer) => {
+          pnpmSpinner.text = data.toString();
```

Comment:
Something like that but maybe bit smarter matching idk?

```suggestion
pnpmSpinner.text = "Progress" + data.toString().split("Progress")[1];
```

Reply:
Yeah, that's what I've just commented haha

Reply:
Astro's was clean and only reused a single line too, how does that work here?

Reply:
Tbf I'm not sure how their code works, I did the same thing they did. In Astro's codebase they also just use the data from the stdout :thinking:

Reply:
> Tbf I'm not sure how their code works, I did the same thing they did. In Astro's codebase they also just use the data from the stdout :thinking: 

Maybe split by `\n`, take the last snd then spöit by the `|`?

Reply:
I used the split by `|` for now, looks as it should I think